fedimint-server/src/lib.rs

```diff
@@ -334,12 +361,19 @@ impl FedimintServer {
         // process messages until new epoch or we have a proposal
         let mut outcomes: Vec<HbbftConsensusOutcome> = loop {
             match self.await_next_epoch().await? {
-                Some(msg) if self.start_next_epoch(&msg) => break self.handle_message(msg).await?,
-                Some(msg) => self.handle_message(msg).await?,
-                None => break vec![],
+                EpochTriggerEvent::NewMessage(msg) if self.start_next_epoch(&msg) => {
+                    break self.handle_message(msg).await?
+                }
+                EpochTriggerEvent::NewMessage(msg) => self.handle_message(msg).await?,
+                _ => break vec![],
             };
         };
 
+        // save any transactions we have in the channel
+        while let Ok(tx) = self.tx_receiver.try_recv() {
+            self.handle_new_tx(Some(tx)).await?;
```

Comment:
This could be a call to `save_transaction_to_db` since no other logic of the `handle_new_tx` function is required here.